ds3-autogen-api/src/main/java/com/spectralogic/ds3autogen/api/models/Ds3Element.java

```diff
@@ -57,4 +59,29 @@ public String getType() {
     public String getComponentType() {
         return componentType;
     }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getName(), getType(), getComponentType());
+    }
+
+    /**
+     * Compares the name, type, and component type. Does not compare the Ds3Annotation list
+     */
+    @Override
+    public boolean equals(final Object obj) {
+        if (obj == this) {
+            return true;
+        }
+        if (!(obj instanceof Ds3Element)) {
+            return false;
+        }
+        final Ds3Element element = (Ds3Element) obj;
+        if (this.getName().equals(element.getName())
```

Comment:
This can probably be simplified to just return the result of the boolean expression